apps/vaporgui/MainForm.cpp

```diff
@@ -1178,6 +1184,11 @@ void MainForm::_fileSaveHelper(string path)
 		return;
 	}
 
+	if (!FileOperationChecker::FileGoodToWrite(fileName)) {
```

Comment:
This "FileGoodToWrite()" check happens AFTER `_controlExec` tries to write the file (L1181). It should, however, happen BEFORE `_controlExec` attempts to write.